### PR TITLE
Use explicit erase to avoid clearing and resizing the hashtable when …

### PR DESCRIPTION
**Pull request by baldersheim**
…usage is small.

@havardpe @geirst @toregge PR

### CODE REVIEW

File: searchlib/src/vespa/searchlib/attribute/changevector.hpp [hunk 1]
```diff
@@ -49,48 +57,32 @@ ChangeVectorT<T>::push_back(uint32_t doc, Accessor & ac)
 {
     if (ac.size() <= 0) { return; }
 
-    size_t index(size());
-    _v.reserve(vespalib::roundUp2inN(index + ac.size(), sizeof(T)));
+    _v.reserve(vespalib::roundUp2inN(size() + ac.size(), sizeof(T)));
```
Comment (havardpe) [inline searchlib/src/vespa/searchlib/attribute/changevector.hpp:60]: consider using new templated variant

**Review by havardpe — APPROVED** [review]

File: searchlib/src/tests/attribute/changevector/changevector_test.cpp [hunk 2]
```diff
@@ -2,17 +2,27 @@
...
+
 template <typename T>
 void verifyStrictOrdering(const T & v) {
-    long count(0);
-    for (const auto & c : v) {
-        count++;
-        EXPECT_EQUAL(count, c._data.get());
+    vespalib::hash_set<uint32_t> complete;
+    uint32_t prev_doc(0);
+    uint32_t prev_value;
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply searchlib/src/tests/attribute/changevector/changevector_test.cpp]: Fixed

File: searchlib/src/vespa/searchlib/attribute/changevector.hpp [hunk 3]
```diff
@@ -12,15 +13,18 @@ namespace {
 // This number is selected to be large enough to hold bursts between commits
 constexpr size_t NUM_ELEMS_TO_RESERVE = 200;
 
+template <typename T>
+constexpr size_t roundUp2inN(size_t elems) {
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply searchlib/src/vespa/searchlib/attribute/changevector.hpp]: Fixed

File: searchlib/src/vespa/searchlib/attribute/changevector.hpp (hunk 1 above)
Comment (baldersheim) [reply searchlib/src/vespa/searchlib/attribute/changevector.hpp]: Fixed

**Pull request merged**